plugins/automation/spec/system/automation_edit_page_spec.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+describe "DiscourseAutomation | Edit page", type: :system do
+  fab!(:admin)
+  fab!(:automation) { Fabricate(:automation, enabled: true) }
+
+  before do
+    DiscourseAutomation::Scriptable.add("required_cats") do
+      field :cat, component: :text, required: true
+      field :dog, component: :text, required: false
+    end
+
+    SiteSetting.discourse_automation_enabled = true
+    sign_in(admin)
+  end
+
+  let(:automation_page) { PageObjects::Pages::Automation.new }
+  let(:tooltip) { PageObjects::Components::Tooltips.new("automation-enabled-toggle") }
+
+  describe "the enabled toggle" do
+    it "can turn on/off the automation" do
+      automation_page.visit(automation)
+
+      expect(automation_page.enabled_toggle).to be_checked
+
+      automation_page.enabled_toggle.toggle
+      automation_page.refresh
```

Comment:
There may be some flakiness here. We are toggling the button but we need to wait for the inflight ajax request to complete before refreshing the page.